include/sc2api/sc2_client.h

```diff
@@ -62,6 +62,10 @@ class ClientEvents {
     //!< \param unit The destroyed unit.
     virtual void OnUnitDestroyed(const Unit*) {}
 
+    //!  Called when a neutral unit is created. For example, mineral fields observed for the first time
+    //!< \param unit The observed unit.
+    virtual void OnNeutralUnitCreated(const Unit*) {}
```

Comment:
To be honest I am not sure whether this event is needed. It could be useful, but requires some time to think about.

Reply:
My bot has a complete map of all of the neutral "snapshot" units at the start of the game. These are used for planning purposes. Bases, for example, have a vector of mineral resources they control. The trouble is, that I can't send a worker to a placeholder, because 1) it isn't possible to interact with it and 2) by the time the worker gets there the unit.tag and (unit *) have changed, so it is in fact a different unit. I use the single notification that the unit is encountered to replace the placeholder unit with an actual unit. Without the notification, it is challenging (older versions of my code had implemented it)
Final note: There are notifications for every other unit: OnUnitEnterVision for enemy and OnUnitCreated for friendly, 
Let me know your thoughts

Reply:
Ok, thank you for the detailed explanation. Let's give it a go.